### PR TITLE
Feature/sniffing 131

### LINKED ISSUE #131: Check (and optionally enable) sniffing of Elasticsearch hosts
All clients that connect to Elasticsearch can use sniffing optionally. Ususally you want it to be enabled but we need to make sure you can deactivate it for some special cases. See https://www.elastic.co/de/blog/elasticsearch-sniffing-best-practices-what-when-why-how for details.

### PR DESCRIPTION
**Pull request by afeefghannam89**
Enable sniffing for logstash
Fix #131

### CODE REVIEW

**Comment by afeefghannam89** [comment]
When I enable sniffing for Kibana with the version 7 I get the error which I set to you Thomas and elasticsearch hosts will not be listed in the log.
`"message":"ResponseError: security_exception: [security_exception] Reason: missing authentication credentials for REST request [/_nodes/_all/http]\n    at onBody (/usr/share/kibana/node_modules/@elastic/elasticsearch/lib/Transport.js:367:23)\n    at IncomingMessage.onEnd (/usr/share/kibana/node_modules/@elastic/elasticsearch/lib/Transport.js:291:11)\n    at IncomingMessage.emit (node:events:525:35)\n    at endReadableNT (node:internal/streams/readable:1358:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)"}`

With version 8.7 I do not see any changes in kibana log or the listed ealsticsearch host.

**Comment by widhalmt** [comment]
From what I found on a short research, the API endpoint Kibana uses for sniffing didn't change from 7.x to 8.x . Maybe the role Kibana uses per default didn't allow sniffing?

Since this only affects 7.x and 7.x won't be supported very long, I could live with disabling sniffing in 7.x completely. We could use `if` in the template and then remove the `if` once 7.x has been removed from the code.

**Comment by afeefghannam89** [comment]
In version 8 I do not see the listed Elasticsearch hosts in logs, when I enable sniffing, they should be logged, should they? In Logstash I can see the listed Elasticsearch hosts after enabling sniffing. In Kibana It is strange!! 
I use the sniffing options in kibana.yml here https://www.elastic.co/guide/en/kibana/current/settings.html

File: roles/elasticsearch/templates/elasticsearch.yml.j2 [hunk 1]
```diff
@@ -2,7 +2,13 @@ node.name: "{{ ansible_hostname }}"
...
 path.logs: {{ elasticsearch_logpath }}
 network.host: ["_local_","_site_"]
-{% if elasticstack_release | int < 8 or groups['elasticsearch'] | length > 1 %}
+{% if ealsticsearch_http_publish_host is defined %}
+http.publish_host: "{{ ealsticsearch_http_publish_host }}"
+{% endif %}
+{% if ealsticsearch_http_publish_port is defined %}
+http.publish_port: {{ ealsticsearch_http_publish_port }}
+{% endif %}
+{% if elastic_release | int < 8 or groups['elasticsearch'] | length > 1 %}
```
Comment (widhalmt) [inline roles/elasticsearch/templates/elasticsearch.yml.j2:11]:
```suggestion
{% if elasticstack_release | int < 8 or groups['elasticsearch'] | length > 1 %}
```

We had to rename this variable due to naming conventrion changes.

File: roles/logstash/templates/elasticsearch-output.conf.j2 [hunk 2]
```diff
@@ -22,7 +22,16 @@ output {
   elasticsearch {
     hosts => [ {% for host in logstash_elasticsearch %}"{{ host }}:{{ elasticstack_elasticsearch_http_port }}"{% if not loop.last %},{% endif %}{% endfor %}]
     validate_after_inactivity => {{ logstash_validate_after_inactivity }}
-{% if elasticstack_full_stack | bool and logstash_security is defined and logstash_security | bool and elasticstack_variant == "elastic" %}
+{% if elastic_stack_full_stack | bool and elastic_variant == "elastic" and logstash_sniffing | bool %}
```
Comment (widhalmt) [inline roles/logstash/templates/elasticsearch-output.conf.j2:25]:
```suggestion
{% if elasticstack_full_stack | bool and elasticstack_variant == "elastic" and logstash_sniffing | bool %}
```
Variables renamed in `main`

File: roles/logstash/templates/elasticsearch-output.conf.j2 [hunk 3]
```diff
@@ -22,7 +22,16 @@ output {
...
+    sniffing => true
+{% if logstash_sniffing_delay is defined %}
+    sniffing_delay => {{ logstash_sniffing_delay }}
+{% if logstash_sniffing_path is defined %}
+    sniffing_path => {{ logstash_sniffing_path }}
+{% endif %}
+{% endif %}
+{% endif %}
+{% if elastic_stack_full_stack | bool and logstash_security is defined and logstash_security | bool and elastic_variant == "elastic" %}
```
Comment (widhalmt) [inline roles/logstash/templates/elasticsearch-output.conf.j2:34]:
```suggestion
{% if elasticstack_full_stack | bool and logstash_security is defined and logstash_security | bool and elastic_variant == "elastic" %}
```

Variable renamed in `main`

**Review by widhalmt — CHANGES_REQUESTED** [review]
Sorry, we had to rename some variables. That broke your changes. I added the new names as "suggestions".

**Comment by afeefghannam89** [comment]
@widhalmt are you ok with the changes here?

**Review by widhalmt — APPROVED** [review]
The full stack variable is still listed as defaulting to `false`. This will be changed globally in https://github.com/NETWAYS/ansible-collection-elasticstack/issues/181 . So yes, I'm very ok with the changes. Thanks. :-)

**Pull request merged**